modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/MatchFunction.java

```diff
@@ -32,7 +33,8 @@
 	@Override
 	public Boolean apply(String value, String regex) {
 		try {
-			Pattern pattern = Pattern.compile(regex, Pattern.DOTALL);
+			Pattern pattern = Pattern.compile(
+				StringUtil.trim(regex), Pattern.DOTALL);
```

Comment:
@linolaoj Not sure if we should change the regex like this. What if the user is intentionally adding a space in the regex? For example: `\w+[ ]\w+`

Reply:
Hi @carolmariaabb , the _trim_ should only remove a space from the beginning and end of the string. The content "inside" the regex should not be affected by the _trim_. I see on the LPS-165388 that the manual test is failing because of extra spaces at the beginning or at the end of the regex, and those spaces are not easy to spot.

Reply:
a regex starting with a white space in Java will always be invalid, if we want the regex to have white space at the beginning (or ending) the safe way would be using "\s".

Reply:
Thanks @linolaoj!